File: EmptyProject/WorkflowsTrainingTwo/WorkflowsTrainingCPlus/Parsing/Expressions/Nodes/Expressions/BooleanLiteralNode.cpp
#include "BooleanLiteralNode.h"
#include <iostream>

namespace WorkflowsTraining::Parsing::Expressions::Nodes::Expressions {
    BooleanLiteralNode::BooleanLiteralNode(bool value)
        : value_(value) {
    }

    std::variant<bool, int, double, std::string> BooleanLiteralNode::Resolve() {
        std::cout << "BooleanLiteralNode: Resolving to " << (value_ ? "true" : "false") << std::endl;
        return value_;
    }
}

File: EmptyProject/WorkflowsTrainingTwo/WorkflowsTraining/Parsing/Expressions/Nodes/Expressions/BooleanLiteralNode.cs
using WorkflowsTraining.Parsing.Expressions.Interfaces;

namespace WorkflowsTraining.Parsing.Expressions.Nodes.Expressions;

public class BooleanLiteralNode(bool value) : IExpressionNode
{
    private bool Value { get; } = value;

    public object Resolve()
    {
        return Value;
    }
}